Makefile

```diff
@@ -15,14 +16,42 @@
 # limitations under the License.
 #
 
-TOPTARGETS := all clean install
+PROJECT_NAME		 := openmme
+VERSION                  ?= $(shell cat ./VERSION)
 
-SUBDIRS := src/common src/s6a src/s1ap src/s11 src/mme-app/
+TOPTARGETS 		 := all clean install
+SUBDIRS 		 := src/common src/s6a src/s1ap src/s11 src/mme-app/
 #$(wildcard src/mme-app/.)
 
 $(TOPTARGETS): $(SUBDIRS)
 $(SUBDIRS):
 	$(MAKE) -C $@ $(MAKECMDGOALS)
 
-.PHONY: $(TOPTARGETS) $(SUBDIRS)
+## Docker related
+DOCKER_REGISTRY          ?=
+DOCKER_REPOSITORY        ?=
+DOCKER_BUILD_ARGS        ?=
+DOCKER_TAG               ?= ${VERSION}
+DOCKER_IMAGENAME         := ${DOCKER_REGISTRY}${DOCKER_REPOSITORY}${PROJECT_NAME}:${DOCKER_TAG}
 
+## Docker labels. Only set ref and commit date if committed
+DOCKER_LABEL_VCS_URL     ?= $(shell git remote get-url $(shell git remote))
+DOCKER_LABEL_VCS_REF     ?= $(shell git diff-index --quiet HEAD -- && git rev-parse HEAD || echo "unknown")
+DOCKER_LABEL_COMMIT_DATE ?= $(shell git diff-index --quiet HEAD -- && git show -s --format=%cd --date=iso-strict HEAD || echo "unknown" )
+DOCKER_LABEL_BUILD_DATE  ?= $(shell date -u "+%Y-%m-%dT%H:%M:%SZ")
+
+# https://docs.docker.com/engine/reference/commandline/build/#specifying-target-build-stage---target
+docker-build:
+	docker build $(DOCKER_BUILD_ARGS) \
+		-t ${DOCKER_IMAGENAME} \
+		--build-arg org_label_schema_version="${VERSION}" \
```

Comment:
Need to add the args in Dockerfile also or not?
@zdw

Reply:
@krsna1729 thanks. done.